c++/src/Compression.cc

```diff
@@ -576,244 +499,201 @@ DIAGNOSTIC_POP
     input->seek(position);
     bytesReturned = static_cast<off_t>(input->ByteCount());
     if (!Skip(static_cast<int>(position.next()))) {
-      throw ParseError("Bad skip in ZlibDecompressionStream::seek");
+      throw ParseError("Bad skip in " + getName());
     }
   }
 
+  class ZlibDecompressionStream : public DecompressionStream {
+  public:
+    ZlibDecompressionStream(std::unique_ptr<SeekableInputStream> inStream,
+                            size_t blockSize,
+                            MemoryPool& pool);
+    virtual ~ZlibDecompressionStream() override;
+    virtual std::string getName() const override;
+
+  protected:
+    virtual void NextDecompress(const void** data,
+                                int* size,
+                                size_t availableSize) override;
+  private:
+    const size_t blockSize;
```

Comment:
This could be removed by using outputDataBuffer.capacity() instead as in BlockDecompressionStream,

Reply:
Done.